akka-actor-typed/src/main/scala/akka/actor/typed/internal/WithMdcBehaviorInterceptor.scala

```diff
@@ -54,7 +54,7 @@ import akka.annotation.InternalApi
           // eliminate that interceptor
           loop(i.nestedBehavior)
 
-        case i: InterceptorImpl[T, T] =>
+        case i: InterceptorImpl[T @unchecked, T @unchecked] =>
```

Comment:
It seems that scala 2.13.7 is catching more warnings. This one was failing at compilation.